lightning/src/ln/channelmanager.rs

```diff
@@ -2428,6 +2452,11 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		let err: Result<(), _> = loop {
 			let mut channel_lock = self.channel_state.lock().unwrap();
 
+			let id = match self.short_to_chan_info.read().unwrap().get(&path.first().unwrap().short_channel_id) {
```

Comment:
e2a0fc8

Reply:
Unfortunately we can't remove the lock order requirement for `short_to_chan_info` to be locked after the `channel_state` lock with this PR, as it's still required in multiple other places in the code, but with #1507 we'll be able to remove that lock order however :).
I think that this order change here and above mainly a preparation for that PR, and is ok here since we only take the `short_to_chan_info`  lock temporarily and drop it before taking the `channel_state` lock.

I agree though that this is a bit confusing, and if you'd like, I could remove this temporary order change.

Reply:
Yes, agree. Once #1772 is merged as well, I'll look into if it makes sense to already change the lock order so that `short_to_chan_info` and `id_to_peer` is before the `channel_state`, as it would be nice to make #1507 less complicated.